File: cpp/utils/TreeNode.cpp
#include <bits/stdc++.h>

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

TreeNode *getSampleRoot() {
    return new TreeNode(
        3,
        new TreeNode(
            4,
            new TreeNode(1),
            new TreeNode(3)),
        new TreeNode(
            5,
            nullptr,
            new TreeNode(1)));
}

void printTree(TreeNode *root) {
    if (!root)
        return;

    std::cout << root->val << " ";
    printTree(root->left);
    printTree(root->right);
}

File: kotlin/TreeNode.kt
class TreeNode(var `val`: Int) {
    var left: TreeNode? = null
    var right: TreeNode? = null
}

fun preOrder(root: TreeNode?) {
    root?.let {
        print("${root.`val`} ")
        preOrder(it.left)
        preOrder(it.right)
    }
}

fun inOrder(root: TreeNode?) {
    root?.let {
        inOrder(it.left)
        print("${root.`val`} ")
        inOrder(it.right)
    }
}
